src/services/tasks/types.ts

```diff
@@ -111,6 +120,30 @@ export interface TasksService {
    */
   cancel(taskId: ObjectId, options?: OptionsBase): Promise<AffectedRecords>;
 
+  /**
+   * ## Execute a Function directly
+   *
+   * A Function may be executed directly, the function will be executed synchronously and the response may be awaited by the caller.
+   *
+   * **Default Permissions:**
+   * - Any party may execute Functions with the `public` permission mode
+   * - Any authenticated user may execute Functions with the `allUsers` permission mode
+   *
+   * **Global Permissions:**
+   * - `EXECUTE_TASK_FUNCTION` - A user may execute all Functions
+   * - `EXECUTE_TASK_FUNCTION:{FUNCTION_NAME}` - A user may execute the Function specified by the FUNCTION_NAME
+   *
+   * @param functionName {@link string} - The functionName property serves as the unique identifier amongst all Functions
+   * @param data {@link U} - The data to be sent to the Function, the type may be user defined
+   * @param options {@link OptionsBase} - Additional options for the request
+   * @returns {@link DirectExecutionResponse} - The response returned from the Function, the response data and results may be user defined
+   */
+  execute<T = any, U = any>(
```

Comment:
This makes me realise I made a mistake in the API Function execution ticket, I didn't give them a default type of any for the generics meaning if you providing typing for the input it will also force you to provide typing for the output.

Not an issue at all, just a consistency thing I should fix at some point.

Reply:
Also unknown and never don't work here. sadge.